src/clj_irods/core.clj

```diff
@@ -308,20 +310,28 @@
       [(fn [cache? irods username zone]
          (when-let [user (if cache?
                            (icat/cached-user irods username zone)
-                           (icat/user irods username zone))]
+                           (when (:has-icat irods)
+                             (icat/user irods username zone)))]
            (delay (if (:user_type_name @user) :user :none)))) username zone] ; for now mimicking jargon version that doesn't distingush
       [(fn [cache? irods username zone]
          (when-let [user (if cache?
                            (jargon/cached-get-user irods username zone)
-                           (jargon/get-user irods username zone))]
+                           (when (:has-jargon irods)
+                             (jargon/get-user irods username zone)))]
            (delay (:type @user)))) username zone])))
 
 (defn uuid->path
   "The path associated with a uuid, or nil."
   [irods uuid]
   (otel/with-span [s ["uuid->path"]]
     (cached-or-get irods
+      [(fn [cache? irods uuid]
+         (if cache?
+           (icat/cached-path-for-uuid irods uuid)
+           (when (:has-icat irods)
+             (icat/path-for-uuid irods uuid)))) uuid]
```

Comment:
This block is the only section in here that's new functionality, the rest is all adding guards to things that depend on jargon or direct ICAT access to their respective flags.